packages/bvaughn-architecture-demo/components/sources/utils/getExpressionFromString.ts

```diff
@@ -1,37 +1,70 @@
+import { highlighter } from "bvaughn-architecture-demo/src/suspense/SyntaxParsingCache";
+
 export default function getExpressionFromString(
-  string: string,
-  characterIndex: number
+  expression: string,
+  cursorIndex: number
 ): string | null {
-  let expression = "";
-  let currentIndex = characterIndex;
-  while (currentIndex >= 0) {
-    const character = string.charAt(currentIndex);
-    if (character === " ") {
-      break;
+  const parsed = highlighter(expression, ".js");
+  if (parsed == null || parsed.length === 0) {
+    return null;
+  } else {
+    const line = parsed[parsed.length - 1];
+
+    const element = document.createElement("div");
+    element.innerHTML = line;
```

Comment:
:speech_balloon: huh. so we're letting Lezer parse things into HTML, and then using an unattached DOM node to figure out what we care about?

That's... hacky, but hey, if it works! :)

Reply:
I agree that it is a bit funky 😁 but we're doing it in a few places and it seems to work well